src/ddb/ddb_vos.c

```diff
@@ -1453,3 +1425,99 @@ dv_clear_committed_table(daos_handle_t coh)
 		return rc;
 	return delete_count;
 }
+
+struct dv_sync_cb_args {
+	dv_smd_sync_complete	 sync_complete_cb;
+	void			*sync_cb_args;
+};
+
+static int
+sync_cb(struct bio_blob_hdr *hdr, void *cb_args)
+{
+	uint8_t			*pool_id = hdr->bbh_pool;
+	struct smd_pool_info	*pool_info = NULL;
+	daos_size_t		 blob_size;
+	uint64_t		 blob_id;
+	int			 rc;
+
+	D_ASSERT(cb_args != NULL);
+	struct dv_sync_cb_args *args = cb_args;
+
+
+	rc = smd_pool_get_info(pool_id, &pool_info);
+	if (!SUCCESS(rc)) {
+		D_ERROR("Failed to get smd pool info: "DF_RC"\n", DP_RC(rc));
+		return rc;
+	}
+
+	/*
+	 * Currently, use the pool's already configured blob size. In the future will need to
+	 * make it more robust and use info from spdk blob's cluster, page size, etc to get
+	 * blob size
+	 */
+	blob_size = pool_info->spi_blob_sz;
+	smd_pool_free_info(pool_info);
+
+	/* Delete the target first if it exists */
+	rc = smd_pool_get_blob(pool_id, hdr->bbh_vos_id, &blob_id);
+	if (rc != DER_NONEXIST) {
```

Comment:
"rc != -DER_NONEXIST"?
On the other hand, is it possible "(rc != 0 && rc != -DER_NONEXIST)", under such case do we need smd_pool_del_tgt()? Why not directly call smd_pool_del_tgt() in spite of whether the target exist or not since we ignore the failuer.